### ISSUE TITLE
[frontend] No metrics shown in Run Output tab

### ISSUE THREAD

**Issue by Kokkini** [kubeflow/pipelines#8356]
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)?
I installed it with the full kubeflow deployment. Kubeflow 1.6, installed with [juju](https://charmed-kubeflow.io/docs/install)
*  KFP version: **build version dev_local** (this shows on bottom of KFP UI left sidenav)

### Steps to reproduce
Running this minimal pipeline that just saves a metrics
```
from kfp.components import OutputPath, create_component_from_func
import kfp.dsl as dsl
import kfp


@dsl.pipeline(
    name='kubeflow-metric-demo')
def metric_pipeline():
    produce_metrics_op = create_component_from_func(
        produce_metrics,
        base_image='python:3.7',
        packages_to_install=[],
        output_component_file='component.yaml',
    )
    produce_metrics_op()


def produce_metrics(
    mlpipeline_metrics_path: OutputPath('Metrics')
):
    import json

    metrics = {
        'metrics': [{
            'name': 'accuracy',
            'numberValue':  0.9,
            'format': "RAW"
        }]
    }
    with open(mlpipeline_metrics_path, 'w') as f:
        json.dump(metrics, f)
    print("metrics path")
    print(mlpipeline_metrics_path)


if __name__ == "__main__":
    kfp.compiler.Compiler().compile(metric_pipeline, 'metric_pipeline.yaml')

```

### Expected result
I expect the metrics to be shown in the "Run output" tab in the UI. However, nothing is there as you can see in the image.
<img width="1192" alt="Screen Shot 2022-10-12 at 3 03 14 PM" src="https://user-images.githubusercontent.com/22306485/195456680-858247f8-9f61-4283-a85d-964c71831de0.png">

However, the metrics did show up at the "output artifacts" section of the component
<img width="1189" alt="Screen Shot 2022-10-12 at 3 07 34 PM" src="https://user-images.githubusercontent.com/22306485/195457044-ea0def8b-0d41-48b5-9734-8927e70b5faf.png">

### Materials and Reference

---

Impacted by this bug? Give it a 👍.

**Comment by Kokkini** [comment]
The metrics is not shown along side each run in the **Experiments** tab either:


<img width="1440" alt="Screen Shot 2022-10-12 at 3 33 16 PM" src="https://user-images.githubusercontent.com/22306485/195460346-01cb5f5c-db9a-4395-9181-b0d66e96c825.png">

**Comment by Lejboelle** [comment]
This has been fixed in the recent patch release [https://github.com/kubeflow/manifests/releases/tag/v1.6.1](url)

**Comment by Kokkini** [comment]
@Lejboelle thank you. I'll try it out.

**Comment by CheaterAdams** [comment]
I too am having this same issue, but with KFP 1.8.16.

**Comment by gkcalat** [comment]
Hi @Kokkini , @CheaterAdams,

A newer release of a platform-agnostic Kubeflow 1.6.1 is [here](https://github.com/kubeflow/manifests/releases/tag/v1.6.1).

**Comment by CheaterAdams** [comment]
If the fix was available in 1.6.1, it is broken in 1.8.16. Can someone tell me what the fix is and are any changes to the above code needed?
thanks

**Comment by CheaterAdams** [comment]
I resolved my problem by using KFP V2 and metrics.log_metric as seen in the code snippet below. With an earlier attempt at using v2 I was not successful. Encountered JSON formatting error. I now believe that was due to not specifying V2 compatible mode when compiling and running from pipeline function. See at the end of this post.


``@component(
    packages_to_install=['scikit-learn'],
    base_image='python:3.9',
)
def digit_classification(metrics: Output[Metrics]):
    
    from sklearn import model_selection
    from sklearn.linear_model import LogisticRegression
    from sklearn import datasets
    from sklearn.metrics import accuracy_score
    from sklearn.metrics import precision_score
...
    #split data
    X_train, X_test, y_train, y_test = model_selection.train_test_split(X, y, test_size=test_size, random_state=seed)
    #fit model
    model.fit(X_train, y_train)

    #accuracy on test set
    result = model.score(X_test, y_test)
    metrics.log_metric('accuracy', (result*100.0))
``

kfp.compiler.Compiler(mode=**kfp.dsl.PipelineExecutionMode.V2_COMPATIBLE**).compile(
    pipeline_func=metrics_visualization_pipeline,
    package_path='pipeline.yaml')

kfp_client.create_run_from_pipeline_func(
    metrics_visualization_pipeline,
    mode=**kfp.dsl.PipelineExecutionMode.V2_COMPATIBLE**,
    # You can optionally override your pipeline_root when submitting the run too:
    # pipeline_root='gs://my-pipeline-root/example-pipeline',
    arguments={})

**Comment by CheaterAdams** [comment]
Hi @Kokkini, does your comment regarding platform agnostic Kubeflow 1.6.1 imply the fix is provided in 1.6.1?
I am still looking for a solution using KPF 1.8.16.

**Comment by michael-EIP** [comment]
Does anyone have solution for this? I'm having this issue also, KFP 1.8.18.

**Comment by ashujain2** [comment]
I am using kubeflow 1.7 with minio hosted on separate namespace (Basically , I have setup minio and kubeflow into same EKS cluster along with Kubeflow with an overall aim to achieve to segregate artifacts and metadata per namespace per bucket )

-> Artifacts and Pipeline logs are going into dedicated buckets
-> Metrics are being generated

Problem - I can not see those metrics on Run Output Tab ,instead of getting metrics on Output artifacts

I am curious to know , How Run Output reads data for metrics to display on UI.

[Commit for above fix ] (https://github.com/jlyaoyuli/pipelines/commit/48c985bf337dbd9e8a8ab7067a05388639e23e21) s merged for kubeflow 1.7

**Comment by yhwang** [comment]
here is the documentation about Run Output: https://www.kubeflow.org/docs/components/pipelines/v1/sdk/output-viewer/
but I don't know the underlying implementation. I was doing the abstract interface implementation if you are curious about that line of code.

**Comment by Ch-Chand** [comment]
### Solution

> Kubeflow 1.8
> KFP SDK 1.8.0

To log metrics in KFP 1.8, you can use the **_mlpipeline_metrics_** output.

```
def modeling(....) -> NamedTuple('Outputs', [('mlpipeline_metrics', 'Metrics')]):

    predictions = model.predict(X_test).flatten()
    rmse = np.sqrt(mean_squared_error(y_test, predictions))
    mae = mean_absolute_error(y_test, predictions)
    r2 = r2_score(y_test, predictions)

    metrics = {
        'metrics': [
            {'name': 'RMSE', 'numberValue': rmse, 'format': 'RAW'},
            {'name': 'MAE', 'numberValue': mae, 'format': 'RAW'},
            {'name': 'R2', 'numberValue': 100 * r2, 'format': 'PERCENTAGE'},
        ]
    }

    return [json.dumps(metrics)]

comp_modeling = kfp.components.create_component_from_func(func=modeling, 
                                                          base_image=modeling_base_image,
                                                          packages_to_install=["pandas", "numpy"])
```

The metrics are logged in a dictionary using the mlpipeline_metrics format. Each metric is represented with a name, its value, and a format (e.g., RAW for numerical values and PERCENTAGE for percentages).

The logged metrics will be visible in the Kubeflow Pipelines UI under each run, allowing you to track and compare performance across runs easily.

**Note:** Solution is feasible for only KFP 1.8
